### PR TITLE
MU WPCOM: Port the newspack blocks from the ETK (2nd try)

### PR DESCRIPTION
**Pull request by arthur791004**
Related to p1722871269505439-slack-CRWCHQGUB

## Proposed changes:

* We merged https://github.com/Automattic/jetpack/pull/38454 before but it had a conflict with the newspack-blocks plugin on atomic sites, so the changes were reverted yesterday.
* Centralize the feature flags of the ETK plugin by the `load_etk_features_flags` function & Load the newspack blocks feature only when the `Newspack_Blocks` isn't declared- https://github.com/Automattic/jetpack/pull/38724/commits/28038374342b793deb6f27efb9cef1225b96cc65
* Load the ETK features as normal but limit the editor-related features to the editor pages or frontend pages to avoid fatal errors when activating the plugin - https://github.com/Automattic/jetpack/pull/38724/commits/603956817518c42bcf09da28a60b96d780489dae

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to your atomic site
* Install the newspack-blocks plugin via https://github.com/Automattic/newspack-blocks/releases/tag/v3.6.0
* Make sure the fatal error won't be shown
  ```
  PHP Fatal error:  Cannot declare class Newspack_Blocks, because the name is already in use in /wordpress/plugins/wpcomsh/5.2.0/vendor/automattic/jetpack-mu-wpcom/src/features/newspack-blocks/synced-newspack-blocks/class-newspack-blocks.php on line 11
  ```

### CODE REVIEW

**Review by taipeicoder — APPROVED** [review]
The changes LGTM, see the conversation in p1722993219555279-slack-CRWCHQGUB for a more in-depth review.

File: projects/packages/jetpack-mu-wpcom/src/features/newspack-blocks/index.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		'newspack_blocks_render_homepage_articles',
+		function () {
+			if ( ! is_admin() ) {
+				\jetpack_mu_wpcom_enqueue_assets( 'newspack-blocks-blog-posts-view', array( 'js', 'css' ) );
+			}
+		}
+	);
+
+	add_action(
+		is_admin() ? 'enqueue_block_assets' : 'enqueue_block_editor_assets',
```
Comment (ellatrix) [inline projects/packages/jetpack-mu-wpcom/src/features/newspack-blocks/index.php:88]: Why did you add these conditions? It should always be `enqueue_block_editor_assets` for block scripts that register the blocks. See #41055.
